src/ng-stomp.js

```diff
@@ -28,14 +28,20 @@ angular
         var dfd = $q.defer()
 
         this.sock = new SockJS(endpoint)
+        this.sock.onclose = function () {
+          if (angular.isFunction(errorCallback)) {
+            errorCallback(new Error('Connection broken'))
```

Comment:
I've tried: when you call `stomp.disconnect()` the `onclose` event is not triggered. So the `errorCallback` will only be called when the connection gets broken or is not established.

Perhaps we can enforce this behavior by setting `this.sock.onclose = null` before calling `this.stomp.disconnect(dfd.resolve)`.